cmd/restic/cmd_check.go

```diff
@@ -172,25 +172,32 @@ func prepareCheckCache(opts CheckOptions, gopts *GlobalOptions) (cleanup func())
 	}
 
 	// use a cache in a temporary directory
-	tempdir, err := os.MkdirTemp(cachedir, "restic-check-cache-")
+	err := os.MkdirAll(cachedir, 0755)
 	if err != nil {
-		// if an error occurs, don't use any cache
-		Warnf("unable to create temporary directory for cache during check, disabling cache: %v\n", err)
+		Warnf("unable to create cachedir %s, disabling cache: %vn", cachedir, err)
 		gopts.NoCache = true
 		return cleanup
-	}
+	} else {
```

Comment:
Instead of `if a { return x; } else { B; }` please write `if a { return x; } B;` (the linter will also complain)